### PR TITLE
added precisions about snapshotting

### PR DESCRIPTION
**Pull request by facchettos**
# Describe what this PR does #

added precisions about snapshotting

#Provide any external context for the change, if any.

provides some precisions about best practices for snapshotting as given in ceph docs and kuebernetes docs 

## Related issues ##

Closes: #1532

### CODE REVIEW

**Comment by facchettos** [comment]
@Madhu-1

File: docs/cephfs-snap-clone.md [hunk 1]
```diff
@@ -35,6 +35,7 @@ kubectl create -f ../examples/cephfs/snapshotclass.yaml
 
 The snapshot is created on/for an existing PVC. You should
 have a PVC in bound state before creating snapshot from it.
+It is recommended to do a snapshot only when the volume is not in use.
```
Comment (pkalever) [inline docs/cephfs-snap-clone.md:38]: Should we tune the terminology here 's/do/create/' and 's/volume/PV/' ?

File: examples/README.md [hunk 2]
```diff
@@ -75,6 +75,8 @@ your Ceph cluster setup.
 If you followed the documentation to create the rbdplugin, you shouldn't
 have to edit any other file.
 
+Note that it is recommended to snapshot only when the volume is not in use.
```
Comment (pkalever) [inline examples/README.md:78]: same here:
` Note that it is recommended to create a snapshot only when the PV is not in use.`

Comment (facchettos) [reply examples/README.md]: Done

File: docs/cephfs-snap-clone.md [hunk 3]
```diff
@@ -35,6 +35,7 @@ kubectl create -f ../examples/cephfs/snapshotclass.yaml
 
 The snapshot is created on/for an existing PVC. You should
 have a PVC in bound state before creating snapshot from it.
+It is recommended to create a snapshot only when the PV is not in use.
```
Comment (Madhu-1) [inline docs/cephfs-snap-clone.md:38]: you also need to add it for clone also. same applies for rbd also
```suggestion
It is recommended to create a volume snapshot/PVC clone only when the PVC is not in use.  
```

**Comment by Madhu-1** [comment]
>mdl --style scripts/mdl-style.rb ./docs/cephfs-snap-clone.md 
./docs/cephfs-snap-clone.md:38: MD009 Trailing spaces
A detailed description of the rules is available at https://github.com/markdownlint/markdownlint/blob/master/docs/RULES.md

CI failure error. https://github.com/ceph/ceph-csi/blob/master/docs/development-guide.md#running-ceph-csi-tests-in-a-container  will help you to test it locally

**Comment by nixpanic** [comment]
Please note that there is a **commitlint** check (part of **continuous-integration/travis-ci/pr**) that requires proper formatted commit messages. Please create a single commit (use `git rebase` and squash all commits together), update the commit message to have a subject prefix of `doc: ` and add your `Signed-off-by:` line (use `git commit --amend`) and `git push --force ...` the changes to the branch for this PR.

You can also run ci/centos jobs (or a standalone **commitlint**) by leaving a comment like:

    /retest ci/centos/containerized-tests

File: docs/cephfs-snap-clone.md [hunk 4]
```diff
@@ -35,6 +35,8 @@ kubectl create -f ../examples/cephfs/snapshotclass.yaml
 
 The snapshot is created on/for an existing PVC. You should
 have a PVC in bound state before creating snapshot from it.
+It is recommended to create a volume snapshot or a PVCclone
```
Comment (Madhu-1) [inline docs/cephfs-snap-clone.md:38]: PVC clone

**Comment by nixpanic** [comment]
Thanks for the updated PR! Just one space needed between `PVCclone` as Madhu commented. Looks good otherwise.

**Comment by facchettos** [comment]
Sorry it took so long to get it right guys

**Comment by Madhu-1** [comment]
> Sorry it took so long to get it right guys

Not at all a problem. the ceph-csi  team will always be happy to help users

**Review by Madhu-1 — APPROVED** [review]

**Review by pkalever — APPROVED** [review]

**Comment by Madhu-1** [comment]
> /retest continuous-integration/travis-ci/pr

I have restarted the Travis CI

**Comment by humblec** [comment]
Thanks @facchettos :+1:

**Review by humblec — APPROVED** [review]

**Pull request merged**